webpack/assets/javascripts/foreman_tools.js

```diff
@@ -85,19 +85,38 @@ export function initTypeAheadSelect(input) {
 // handle table updates via turoblinks
 export function updateTable(element) {
   let uri = new URI(window.location.href);
-  let search, perPage;
+  let searchTerm, perPage, isSearchForm;
 
   if (element !== undefined) {
-    search = $(element).find('.autocomplete-input').val() || $('#search-form').find('.autocomplete-input').val();
-    if (search !== undefined) {
-      uri.setSearch('search', search.trim());
+    isSearchForm = (element.id === 'search-form');
+    if ((isSearchForm || element.id === 'per_page') && getQueryStringValue(uri, 'page') > 1) {
+      uri.setSearch('page', 1);
+    }
+    if (isSearchForm) {
+      searchTerm = $(element).find('.autocomplete-input').val() || $('#search-form').find('.autocomplete-input').val();
+      if (searchTerm) {
+        uri.setSearch('search', searchTerm.trim());
+      }
+    }
+    perPage = $('#per_page').val();
+    if (perPage !== undefined) {
+      if (element.id !== 'search-form' && !(uri.hasSearch('search'))) {
```

Comment:
the first clause here is the same as `!isSearchForm`, and the second I think is redundant - if the uri doesn't have search, there is nothing to remove.